src/Datadog.Trace/Tracer.cs

```diff
@@ -118,7 +118,26 @@ internal Tracer(TracerSettings settings, IAgentWriter agentWriter, ISampler samp
                 });
 
             // fall back to default implementations of each dependency if not provided
-            _agentWriter = agentWriter ?? new AgentWriter(new Api(Settings.AgentUri, apiRequestFactory: null, Statsd), Statsd);
+            if (agentWriter != null)
+            {
+                _agentWriter = agentWriter;
+            }
+            else
+            {
+                IApi api = null;
+                switch (Settings.Exporter)
+                {
+                    case ExporterType.Zipkin:
+                        api = new ZipkinApi(Settings);
```

Comment:
I'm passing the settings to have access to the agent URL: if there is an agreement I think this should be changed to the sources so each exporter (or other components) can do their own configuration reading.

Reply:
Your suggestion makes sense, and matches the pattern used within the OTel SDK.

Reply:
Yeah that's a good suggestion 👍🏼

Reply:
I think we have an agreement here. Let's tackle that in a separate review.